source/slang/slang-capability.cpp

```diff
@@ -127,6 +127,31 @@ CapabilityAtom findCapabilityAtom(UnownedStringSlice const& name)
     return CapabilityAtom::Invalid;
 }
 
+bool isCapabilityDerivedFrom(CapabilityAtom atom, CapabilityAtom base)
```

Comment:
This is good to have, but I believe the capability-set stuff also has support for querying a compacted/filtered version of a capability set. That set should only include atoms that aren't already implied by other atoms in the set (e.g., it should not include both `spirv_1_3` and `spirv_1_4` because the latter implies the former). It might be enough to use that filtering logic and then directly enumerate the result knowing it won't have any atoms that should be skipped.

(I'll have to look at the actual logic later to see, and it is possible that I'm imagining that I actually implemented stuff in `CapabilitySet`)